UnityProject/Assets/Scripts/Messages/Server/TabUpdateMessage.cs

```diff
@@ -55,68 +108,178 @@ public static void SendToPeepers(GameObject provider, NetTabType type, TabAction
 		ElementValue[] values = null)
 	{
 
-		if (values != null)
+		switch (tabAction)
+		{
+			case TabAction.Open:
+				NetworkTabManager.Instance.Add(provider, type, recipient);
+				values = NetworkTabManager.Instance.Get(provider, type).ElementValues;
+				break;
+			case TabAction.Close:
+				NetworkTabManager.Instance.Remove(provider, type, recipient);
+				break;
+			case TabAction.Update:
+				//fixme: duplication of NetTab.ValidatePeepers
+				//Not sending updates and closing tab for players that don't pass the validation anymore
+				var validate = Validations.CanApply(recipient, provider, NetworkSide.Server);
+				if (!validate)
+				{
+					Send(recipient, provider, type, TabAction.Close);
+					return null;
+				}
+				break;
+		}
+
+		// SingleMessage: 0, start/middle: 1, end: 2
+		var id = 0;
+
+		var uniqueID = Random.Range(0, 2000);
```

Comment:
that's certainly not unique lol

Reply:
It only has to be unique for the sending, then as soon as its sent can be reused. The chance it fails both checks is minuscule, so i dont really expect collisions.

Reply:
But yeah i'll see about changing it to a counter.